samples/matter/lock/src/bolt_lock_manager.cpp

```diff
@@ -20,118 +20,52 @@ void BoltLockManager::Init(StateChangeCallback callback)
-				    EmberAfPluginDoorLockCredentialInfo &credential) const
+				    EmberAfPluginDoorLockCredentialInfo &credential)
 {
-	VerifyOrReturnError(credentialIndex > 0 && credentialIndex <= CONFIG_LOCK_NUM_CREDENTIALS, false);
-
-	credential = mCredentials[credentialIndex - 1];
-
-	ChipLogProgress(Zcl, "Getting lock credential %u: %s", static_cast<unsigned>(credentialIndex),
-			credential.status == DlCredentialStatus::kAvailable ? "available" : "occupied");
-
-	return true;
+	return PinManager::Instance().GetCredential(credentialIndex, credentialType, credential);
 }
 
 bool BoltLockManager::SetCredential(uint16_t credentialIndex, FabricIndex creator, FabricIndex modifier,
 				    DlCredentialStatus credentialStatus, CredentialTypeEnum credentialType,
 				    const ByteSpan &secret)
 {
-	VerifyOrReturnError(credentialIndex > 0 && credentialIndex <= CONFIG_LOCK_NUM_CREDENTIALS, false);
-	VerifyOrReturnError(secret.size() <= kMaxCredentialLength, false);
-
-	CredentialData &credentialData = mCredentialData[credentialIndex - 1];
-	auto &credential = mCredentials[credentialIndex - 1];
-
-	if (!secret.empty()) {
-		memcpy(credentialData.mSecret.Alloc(secret.size()).Get(), secret.data(), secret.size());
-	}
-
-	credential.status = credentialStatus;
-	credential.credentialType = credentialType;
-	credential.credentialData = ByteSpan(credentialData.mSecret.Get(), secret.size());
-	credential.creationSource = DlAssetSource::kMatterIM;
-	credential.createdBy = creator;
-	credential.modificationSource = DlAssetSource::kMatterIM;
-	credential.lastModifiedBy = modifier;
-
-	ChipLogProgress(Zcl, "Setting lock credential %u: %s", static_cast<unsigned>(credentialIndex),
-			credential.status == DlCredentialStatus::kAvailable ? "available" : "occupied");
-
-	return true;
+	return PinManager::Instance().SetCredential(credentialIndex, creator, modifier, credentialStatus,
+						    credentialType, secret);
 }
 
-bool BoltLockManager::ValidatePIN(const Optional<ByteSpan> &pinCode, OperationErrorEnum &err) const
+bool BoltLockManager::ValidatePIN(const Optional<ByteSpan> &pinCode, OperationErrorEnum &err)
 {
-	/* Optionality of the PIN code is validated by the caller, so assume it is OK not to provide the PIN code. */
-	if (!pinCode.HasValue()) {
-		return true;
-	}
-
-	/* Check the PIN code */
-	for (const auto &credential : mCredentials) {
-		if (credential.status == DlCredentialStatus::kAvailable ||
-		    credential.credentialType != CredentialTypeEnum::kPin) {
-			continue;
-		}
-
-		if (credential.credentialData.data_equal(pinCode.Value())) {
-			ChipLogDetail(Zcl, "Valid lock PIN code provided");
-			return true;
-		}
-	}
-
-	ChipLogDetail(Zcl, "Invalid lock PIN code provided");
-	err = OperationErrorEnum::kInvalidCredential;
+	return PinManager::Instance().ValidatePIN(pinCode, err);
+}
 
-	return false;
+void BoltLockManager::SetRequirePIN(bool require)
+{
+	return PinManager::Instance().SetRequirePIN(require);
```

Comment:
```suggestion
	PinManager::Instance().SetRequirePIN(require);
```
It returns void anyway.

Reply:
Yes, but I guess it's a good practice to forward the function call by return because the return types of the caller and callee are in line. If the signature of the SetRequirePIN() changes and it suddenly starts returning non-void the compiler will react.